webapp/components/TeaserImage/TeaserImage.vue

```diff
@@ -1,257 +0,0 @@
-        this.error = false
-      }, 2000)
-    },
-    transformImage(file) {
-      this.file = file
-      this.$emit('cropInProgress', true)
-      this.showCropper = true
-      this.initEditor()
-      this.initCropper()
-    },
-    initEditor() {
-      this.editor = this.$refs.cropperOverlay
-      this.clearImages()
-      this.thumbnailElement.appendChild(this.editor)
-    },
-    clearImages() {
-      this.thumbnailElement = document.querySelectorAll('#postdropzone')[0]
-      const thumbnailPreview = document.querySelectorAll('.thumbnail-preview')[0]
-      if (thumbnailPreview) thumbnailPreview.remove()
-      const contributionImage = document.querySelectorAll('.contribution-image')[0]
-      this.oldImage = contributionImage
-      if (contributionImage) contributionImage.remove()
-    },
-    deleteImage() {
-      this.clearImages()
-    },
-    initCropper() {
-      this.image = new Image()
-      this.image.src = URL.createObjectURL(this.file)
-      this.editor.appendChild(this.image)
-      this.cropper = new Cropper(this.image, { zoomable: false, autoCropArea: 0.9 })
-    },
-    cropImage() {
-      this.showCropper = false
-      if (this.file.type === 'image/jpeg') {
-        this.uploadJpeg()
-      } else {
-        this.uploadOtherImageType()
-      }
-    },
-    uploadOtherImageType() {
-      this.imageAspectRatio = this.file.width / this.file.height || 1.0
-      this.image = new Image()
-      this.image.src = this.file.dataURL
-      this.setupPreview()
-      this.emitImageData(this.file)
-    },
-    uploadJpeg() {
-      const canvas = this.cropper.getCroppedCanvas()
-      canvas.toBlob(blob => {
-        this.imageAspectRatio = canvas.width / canvas.height
-        this.image = new Image()
-        this.image.src = canvas.toDataURL()
-        this.setupPreview()
-        const croppedImageFile = new File([blob], this.file.name, { type: this.file.type })
-        this.emitImageData(croppedImageFile)
-      }, 'image/jpeg')
-    },
-    setupPreview() {
-      this.image.classList.add('thumbnail-preview')
-      this.thumbnailElement.appendChild(this.image)
-    },
-    cancelCrop() {
-      if (this.oldImage) this.thumbnailElement.appendChild(this.oldImage)
-      this.showCropper = false
-      this.$emit('cropInProgress', false)
-    },
-    emitImageData(imageFile) {
-      this.$emit('addTeaserImage', imageFile)
-      this.$emit('addImageAspectRatio', this.imageAspectRatio)
-      this.$emit('cropInProgress', false)
-    },
-  },
-}
-</script>
-<style lang="scss">
-#postdropzone {
-  width: 100%;
-  min-height: 400px;
-  background-color: $background-color-softest;
-}
-
-.hc-attachments-upload-area-post {
-  position: relative;
-  display: flex;
-  justify-content: center;
-  cursor: pointer;
-}
-
-.hc-attachments-upload-area-update-post img {
-  object-fit: cover;
-  object-position: center;
-  display: block;
-  width: 100%;
-}
-
-.hc-attachments-upload-area-update-post:hover {
```

Comment:
> [<img alt="mattwr18" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/mattwr18) **Authored by [mattwr18](https://github.com/mattwr18)**
_<time datetime="2020-02-12T13:26:46Z" title="Wednesday, February 12th 2020, 2:26:46 pm +01:00">Feb 12, 2020</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -58,181 +49,142 @@ export default {
       dropzoneOptions: {
         url: () => '',
         maxFilesize: 5.0,
-        previewTemplate: this.template(),
+        previewTemplate: '<img class="preview-image" />',
       },
-      image: null,
-      file: null,
-      editor: null,
       cropper: null,
-      thumbnailElement: null,
-      oldImage: null,
-      error: false,
+      file: null,
       showCropper: false,
+      isLoadingImage: false,
     }
   },
   methods: {
-    template() {
-      return `<div class="dz-preview dz-file-preview">
-                <div class="dz-image">
-                  <div data-dz-thumbnail-bg></div>
-                </div>
-              </div>
-	     `
-    },
-    verror(file, message) {
-      this.error = true
+    onDropzoneError(file, message) {
       this.$toast.error(file.status, message)
-      setTimeout(() => {
-        this.error = false
-      }, 2000)
-    },
-    transformImage(file) {
-      this.file = file
-      this.showCropper = true
-      this.initEditor()
-      this.initCropper()
-    },
-    initEditor() {
-      this.editor = this.$refs.cropperOverlay
-      this.clearImages()
-      this.thumbnailElement.appendChild(this.editor)
     },
     clearImages() {
-      this.thumbnailElement = document.querySelectorAll('#postdropzone')[0]
-      const thumbnailPreview = document.querySelectorAll('.thumbnail-preview')[0]
-      if (thumbnailPreview) thumbnailPreview.remove()
-      const contributionImage = document.querySelectorAll('.contribution-image')[0]
-      this.oldImage = contributionImage
-      if (contributionImage) contributionImage.remove()
+      const images = document.querySelectorAll('.preview-image')
+      images.forEach((image, index) => {
+        if (index === images.length - 1) image.src = ''
+        else image.remove()
+      })
     },
-    initCropper() {
-      this.image = new Image()
-      this.image.src = URL.createObjectURL(this.file)
-      this.editor.appendChild(this.image)
-      this.cropper = new Cropper(this.image, { zoomable: false, autoCropArea: 0.9 })
+    initCropper(file) {
+      this.showCropper = true
+      this.file = file
+      this.clearImages()
+
+      const imageElement = document.querySelector('#cropping-image')
+      imageElement.src = URL.createObjectURL(file)
+      this.cropper = new Cropper(imageElement, { zoomable: false, autoCropArea: 0.9 })
     },
     cropImage() {
-      this.showCropper = false
+      this.isLoadingImage = true
       const canvas = this.cropper.getCroppedCanvas()
       canvas.toBlob(blob => {
         const imageAspectRatio = canvas.width / canvas.height
-        this.setupPreview(canvas)
-        this.removeCropper()
         const croppedImageFile = new File([blob], this.file.name, { type: this.file.type })
         this.$emit('addTeaserImage', croppedImageFile)
         this.$emit('addImageAspectRatio', imageAspectRatio)
+        this.setupPreview(canvas.toDataURL())
       }, 'image/jpeg')
+
+      this.closeCropper()
     },
-    setupPreview(canvas) {
-      this.image = new Image()
-      this.image.src = canvas.toDataURL()
-      this.image.classList.add('thumbnail-preview')
-      this.thumbnailElement.appendChild(this.image)
+    setupPreview(url) {
+      const previewElement = document.querySelector('.image-uploader .preview-image')
+      previewElement.src = url
+      this.$nextTick((this.isLoadingImage = false))
     },
-    cancelCrop() {
+    closeCropper() {
       this.showCropper = false
-      if (this.oldImage) this.thumbnailElement.appendChild(this.oldImage)
-      this.removeCropper()
-    },
-    removeCropper() {
-      this.editor.removeChild(document.querySelectorAll('.cropper-container')[0])
+      this.cropper.destroy()
```


I'm really curious to see this all work! I really like the refactor! and hope it behaves the same.
This is where I'm kicking myself for not adding cypress tests, so that we could make sure it still does.
Anyways, I'll manually test it!

Reply:
> [<img alt="alina-beck" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/alina-beck) **Authored by [alina-beck](https://github.com/alina-beck)**
_<time datetime="2020-02-17T10:02:41Z" title="Monday, February 17th 2020, 11:02:41 am +01:00">Feb 17, 2020</time>_

---


Thank you for taking the time to test it! This is really the part where I was most scared to break something...